### PR TITLE
[NODE-1015] Added ability to parse compression options

### PR DESCRIPTION
**Pull request by malexandert**

### CODE REVIEW

File: test/functional/url_parser_tests.js [hunk 1]
```diff
@@ -426,7 +426,27 @@ exports['Parse mongodb://localhost/db?w=-1'] = {
 /**
  * @ignore
  */
-exports['Throw on unsuported options'] = {
+exports['Should be able to parse mongodb://localhost/?compressors=snappy,zlib&zlibCompressionLevel=-1'] = {
```
Comment (mbroadst) [inline test/functional/url_parser_tests.js]: What about these cases:
  - one only compressor specified
  - `zlibCompressionLevel` specified with no compressors
  - `zlibCompressionLevel` specified with `snappy` compression

I know we discussed that the last two are probably no-ops (due to spec), but let's put our assumptions to paper here.

File: test/functional/mongo_client_tests.js [hunk 2]
```diff
@@ -436,6 +436,20 @@ exports["correctly error out when no socket available on MongoClient.connect"] =
   }
 }
 
+exports["correctly handles compression"] = {
```
Comment (mbroadst) [inline test/functional/mongo_client_tests.js]: I'm not sure this test is quite useful atm, unless you can validate that the internal state object was created with the expected compression options. I imagine you will have far more comprehensive tests soon that actually verify that the expected compression is used based on URI (_actually_ testing the compression etc)

File: test/functional/url_parser_tests.js (hunk 1 above)
Comment (mbroadst) [inline test/functional/url_parser_tests.js]: nitpick: I think we can name this test better. I know you've used the same form used elsewhere in the document here, but I think it's better to clearly indicate what part of uri parsing you're actually testing here.

**Comment by malexandert** [comment]
Tagging @sebthedev so he also gets emails (I think this will work?)

**Comment by malexandert** [comment]
@mbroadst ready for another round of review!

**Pull request merged**